### PR TITLE
Users registration API

### LINKED ISSUE #5: users registration API
This registration API accepts the form data for individuals and companies. 

- Check the data for company or individual business

- Clean up data

- Validate data

- Store in MongoDB Users collection

### PR DESCRIPTION
**Pull request by nextwebb**
feat: allow individuals and companies to register with proper validation and generate Paytax id.
functions include:

-  server-side validation of data inputs,

- error messages

- generation of tax payer

- email registered users their Paytax Id

- update db users collection

Fixes #5 #6 #7

### CODE REVIEW

File: src/controllers/usersController.js [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+exports.registeration = (req, res) => {
+  const user = new User(req.body);
+  user.register().then((result) => {
+    const msg = {
+      to: `${result.email}`,
+      from: 'contact@app.paytax.com',
+      subject: 'PayTax APP registration',
+      text: 'Pay your tax anywhere with app.paytax.com',
+      html: `<strong> Hi 😀 ${result.businessName} we've received your registration; Here's your Tax payer ID: <h1> ${result.taxPayerId}</h1>  </strong>`
+    };
```
Comment (triple0t) [inline src/controllers/usersController.js]: You might have to abstract this into its own method

File: src/models/Users.js [hunk 2]
```diff
@@ -1,42 +1,176 @@
...
+  if (this.data.company === 'true') {
+    // company account reg
+    if (typeof (this.data.name) !== 'string') { this.data.name = ''; }
+    if (typeof (this.data.address) !== 'string') { this.data.address = ''; }
+    if (typeof (this.data.email) !== 'string') { this.data.email = ''; }
+    if (typeof (this.data.phone) !== 'string') { this.data.phone = ''; }
+    if (typeof (this.data.bvn) !== 'string') { this.data.bvn = ''; }
+    if (typeof (this.data.password) !== 'string') { this.data.password = ''; }
+    if (typeof (this.data.state) !== 'string') { this.data.state = ''; }
+    if (typeof (this.data.city) !== 'string') { this.data.city = ''; }
```
Comment (triple0t) [inline src/models/Users.js]: build a reusable method for this. the DRY rule in programming

File: src/models/Users.js [hunk 3]
```diff
@@ -1,42 +1,176 @@
...
+  }
+  return result;
+};
+
+User.prototype.login = function () {
+  return new Promise((resolve, reject) => {
+
+  });
+};
+
```
Comment (triple0t) [inline src/models/Users.js:130]: Kindly comment your code please

File: src/models/Users.js (hunk 2 above)
Comment (nextwebb) [reply src/models/Users.js]: Alright

**Review by Ebugo — DISMISSED** [review]
Did you try using Express validator? I think it makes the validation codes in all more compact.

**Review by nextwebb — COMMENTED** [review]
> Did you try using Express validator? I think it makes the validation codes in all more compact.
i used the validator package.

File: src/models/Users.js [hunk 4]
```diff
@@ -1,42 +1,179 @@
...
+    this.cleanUp();
+    await this.validate();
+    this.data.taxPayerId = await User.makeId(10);
+    if (!this.errors.length) {
+      // hash user password
+      const salt = bcrypt.genSaltSync(10);
+      this.data.password = bcrypt.hashSync(this.data.password, salt);
+      // upload to users collection
+      await usersCollection.insertOne(this.data);
+
```
Comment (triple0t) [inline src/models/Users.js:157]: Remember to wrap these in a try-catch block

**Review by triple0t — APPROVED** [review]
LGTM 🚀

**Review by Ebugo — APPROVED** [review]

**Pull request merged**